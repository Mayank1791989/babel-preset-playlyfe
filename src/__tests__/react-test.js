/* @flow */
import { testParseCode, transform } from './test-utils';

testParseCode('react jsx', {
  opts: [null, undefined, { react: true }],
  throws: false,
  code: `
    class Component extends React.Component {
      render() {
        return (
          <div>
            <span>Test</span>
          </div>
        );
      }
    }
  `,
});

testParseCode('react jsx', {
  opts: [{ react: false }],
  throws: true,
  code: `
    class Component extends React.Component {
      render() {
        return (
          <div>
            <span>Test</span>
          </div>
        );
      }
    }
  `,
});

test('include react source and self for dev', () => {
  const code = `
    const test = () => {
      return <div />;
    }
  `;
  const transformed = transform(code, { react: true });
  expect(transformed.code).toMatchSnapshot();
});
