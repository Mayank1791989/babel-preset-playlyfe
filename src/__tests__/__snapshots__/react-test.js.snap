// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`support react-intl correctly throw invalid intl message error 1`] = `
"/test-file.js: [React Intl] Message failed to parse. See: http://formatjs.io/guides/message-syntax/
SyntaxError: Expected \\"'\\", \\"=\\", \\"\\\\\\\\#\\", \\"\\\\\\\\\\\\\\\\\\", \\"\\\\\\\\u\\", \\"\\\\\\\\{\\", \\"\\\\\\\\}\\", \\"}\\", or [^{}\\\\\\\\\\\\0-\\\\x1F\\\\x7F \\\\t\\\\n\\\\r] but end of input found.
   6 |             <FormattedMessage
   7 |               id=\\"test.xyz\\"
>  8 |               defaultMessage=\\"{xyz, select, y {test}\\"
     |                              ^
   9 |             />
  10 |           );
  11 |         "
`;
